dirsrvtests/tests/suites/password/pwdPolicy_attribute_test.py

```diff
@@ -300,6 +301,618 @@ def test_pwd_min_age(topology_st, create_user, password_policy):
+    # passwordOTPMaxUse so that OTP is enforced first
+    topology_st.standalone.config.replace('passwordMustChange', 'on')
+    topology_st.standalone.config.replace('passwordOTPDelayValidFrom', str(ValidFrom))
+    time.sleep(.5)
+
+    # Reset user's password
+    our_user = UserAccount(topology_st.standalone, TEST_USER_DN)
+    our_user.replace('userpassword', PASSWORD)
+    # give time to update the pwp attributes in the entry
+    time.sleep(.5)
+
+    # Check that pwdReset is TRUE
+    topology_st.standalone.simple_bind_s(DN_DM, PASSWORD)
+    assert our_user.get_attr_val_utf8('pwdReset') == 'TRUE'
+
+    # Check that pwdOTPReset is TRUE
+    assert our_user.get_attr_val_utf8('pwdOTPReset') == 'TRUE'
+    now = time.mktime(time.gmtime())
+    log.info("compare pwdOTPValidFrom (%s) vs now (%s)" % (our_user.get_attr_val_utf8('pwdOTPValidFrom'), time.gmtime()))
+    assert (gentime_to_posix_time(our_user.get_attr_val_utf8('pwdOTPValidFrom'))) >= (now + ValidFrom - 2)
+
+    # wait for pwdOTPValidFrom
+    time.sleep(ValidFrom + 1)
+
+    # Bind as user with valid password, reset the password
+    # and do simple search
+    our_user.rebind(PASSWORD)
+    our_user.reset_password(TEST_USER_PWD)
+    our_user.rebind(TEST_USER_PWD)
+    assert our_user.get_attr_val_utf8('uid')
+
+    def fin():
+        topology_st.standalone.restart()
+        # Reset password policy config
+        topology_st.standalone.simple_bind_s(DN_DM, PASSWORD)
+        topology_st.standalone.config.replace('passwordMustChange', 'off')
+
+        # Reset user's password
+        our_user.replace('userpassword', TEST_USER_PWD)
+
+    request.addfinalizer(fin)
+
+def test_global_otp_delayExpireAt_1(topology_st, create_user, request):
+    """Test global OTP policy : passwordOTPDelayExpireAt
+    Test that a OTP password is not valid after reset time +
+    passwordOTPDelayExpireAt
+
+    :id: b98def32-4e30-49fd-893b-8f959ba72b98
+    :customerscenario: False
+    :setup: Standalone instance
+    :steps:
+        1. Enable passwordMustChange
+        2. Set passwordOTPDelayExpireAt=6s
+        3. Create a account user
+        5. Reset the password
+        6. Wait for passwordOTPDelayExpireAt=6s + 2s (safety)
+        7. Bind with valid password should fail with ldap.CONSTRAINT_VIOLATION
+    :expected results:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+    """
+
+    ExpireAt = 6
+    # Set password policy config, passwordMaxFailure being higher than
+    # passwordOTPMaxUse so that OTP is enforced first
+    topology_st.standalone.config.replace('passwordMustChange', 'on')
+    topology_st.standalone.config.replace('passwordOTPMaxUse', str(-1))
+    topology_st.standalone.config.replace('passwordOTPDelayValidFrom', str(-1))
+    topology_st.standalone.config.replace('passwordOTPDelayExpireAt', str(ExpireAt))
```

Comment:
I'm not sure about the naming of DelayExpireAt and how it's working. Is this acting as a default offset of the expireAt value if one is not set?

Reply:
'passwordOTPDelayxxx' are offsets (in second) from timestamp when the password reset occurs (admin setting userpassword). Those offsets define a period of time when the userpassword is accepted for authentication.
For example, let a policy with passwordOTPDelayValidFrom=60, passwordOTPDelayExpireAt=3600 and userpassword being reset at 14:05. The end user can authenticate with the userpassword between 14:06 and 15:05.
The defaults of passwordOTPDelayxxx is '-1', it means that this limit of the window is not enforced.